# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/indexedDB/IDBTransaction.cpp
# Commit: fc5e1bee1332
# Full Hash: fc5e1bee13321946704a915ef4c7c80c0941dcb3
# Author: Simon Giesecke <sgiesecke@mozilla.com>
# Date: 2019-10-02 03:38:52
# Regressor Bug: 1168606
# File Overlap Count: 1
# Description:
#   Bug 1168606 - Send two records with every ObjectStoreCursorResponse. r=ttung,asuth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D43252
# ==============================================================================

diff -r 63cf0966cb41 -r fc5e1bee1332 dom/indexedDB/IDBTransaction.cpp
--- a/dom/indexedDB/IDBTransaction.cpp	Tue Oct 01 09:15:47 2019 +0000
+++ b/dom/indexedDB/IDBTransaction.cpp	Tue Oct 01 09:15:49 2019 +0000
@@ -338,7 +338,7 @@
   ++mPendingRequestCount;
 }
 
-void IDBTransaction::OnRequestFinished(bool aActorDestroyedNormally) {
+void IDBTransaction::OnRequestFinished(bool aRequestCompletedSuccessfully) {
   AssertIsOnOwningThread();
   MOZ_ASSERT(mPendingRequestCount);
 
@@ -347,7 +347,7 @@
   if (!mPendingRequestCount) {
     mReadyState = COMMITTING;
 
-    if (aActorDestroyedNormally) {
+    if (aRequestCompletedSuccessfully) {
       if (NS_SUCCEEDED(mAbortCode)) {
         SendCommit();
       } else {
@@ -797,6 +797,27 @@
   return mNextIndexId++;
 }
 
+void IDBTransaction::InvalidateCursorCaches() {
+  AssertIsOnOwningThread();
+
+  for (auto* const cursor : mCursors) {
+    cursor->InvalidateCachedResponses();
+  }
+}
+
+void IDBTransaction::RegisterCursor(IDBCursor* const aCursor) {
+  AssertIsOnOwningThread();
+
+  mCursors.AppendElement(aCursor);
+}
+
+void IDBTransaction::UnregisterCursor(IDBCursor* const aCursor) {
+  AssertIsOnOwningThread();
+
+  DebugOnly<bool> removed = mCursors.RemoveElement(aCursor);
+  MOZ_ASSERT(removed);
+}
+
 nsIGlobalObject* IDBTransaction::GetParentObject() const {
   AssertIsOnOwningThread();
 